src/hypothesis/core.py

```diff
@@ -142,16 +146,26 @@ def do_draw(self, data):
 
 
 def is_invalid_test(
-    name, original_argspec, generator_arguments, generator_kwargs
+    name, original_argspec, type_hints, generator_arguments, generator_kwargs
 ):
-    def invalid(message):
+    def invalid(message, exception=InvalidArgument):
         def wrapped_test(*arguments, **kwargs):
-            raise InvalidArgument(message)
+            raise exception(message)
         return wrapped_test
 
     if not (generator_arguments or generator_kwargs):
-        return invalid(
-            'given must be called with at least one argument')
+        if not type_hints:
+            return invalid(
+                '@given must be called with at least one argument, or used '
+                'to wrap a test with type hints.'
+            )
+        elif all(v.is_empty for v in type_hints.values()):
```

Comment:
I don't think I like this behaviour of using `is_empty` as an error condition for `from\_type`. I would much rather it raise an exception when called with an unresolvable type.